scripts/download-sponsors.js

```diff
@@ -6,60 +6,189 @@ const fs = require("fs");
+const transactionsGraphqlQuery = `query transactions($dateFrom: ISODateTime, $limit: Int, $offset: Int) {
+  transactions(account: {
+    slug: "babel"
+  }, dateFrom: $dateFrom, limit: $limit, offset: $offset, includeIncognitoTransactions: false) {
+    nodes {
+        amountInHostCurrency {
+          value
         }
         fromAccount {
           name
           slug
           website
           imageUrl
-          description
         }
+        createdAt
       }
-    }
   }
 }`;
 
-const sponsorsFile = `${__dirname}/../website/data/sponsors.json`;
-
-console.log("Downloading sponsors data from Open Collective...");
-
-fetch(graphqlEndpoint, {
-  method: "POST",
-  headers: { "Content-Type": "application/json" },
-  body: JSON.stringify({ query: graphqlQuery }),
-})
-  .then(res => res.json())
-  .then(res => res.data.account.orders.nodes)
-  .then(nodes =>
-    nodes
-      .filter(node => !!node.tier)
-      .map(node => ({
-        tier: node.tier.slug,
-        name: node.fromAccount.name,
-        slug: node.fromAccount.slug,
-        website: node.fromAccount.website,
-        avatar: node.fromAccount.imageUrl,
-        twitterHandle: node.fromAccount.twitterHandle,
-        description: node.fromAccount.description,
-      }))
-  )
-  .then(sponsors => JSON.stringify(sponsors, null, 2))
-  .then(sponsorsJson => {
-    fs.writeFile(sponsorsFile, sponsorsJson, err => {
-      if (err) {
-        console.error("Failed to write website/data/sponsors.json file: ", err);
-      } else {
-        console.log("Success: website/data/sponsors.json created.");
+const nodeToSupporter = node => ({
+  name: node.account.name,
+  slug: node.account.slug,
+  website: node.account.website,
+  avatar: node.account.imageUrl,
+  firstDonation: node.createdAt,
+  totalDonations: node.totalDonations.value,
+  yearlyDonations: 0,
+  monthlyDonations: 0,
+});
+
+const getAllNodes = async (graphqlQuery, getNodes, time = "year") => {
+  const body = {
+    query: graphqlQuery,
+    variables: {
+      limit: graphqlPageSize,
+      offset: 0,
+      dateFrom: new Date(
+        time === "year"
+          ? new Date().setFullYear(new Date().getFullYear() - 1) // data from last year
+          : new Date().setMonth(new Date().getMonth() - 1) // data from last month
+      ).toISOString(),
+    },
+  };
+
+  let allNodes = [];
+
+  // Handling pagination if necessary
+  // eslint-disable-next-line
+  while (true) {
+    const result = await fetch(graphqlEndpoint, {
+      method: "POST",
+      body: JSON.stringify(body),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    }).then(response => response.json());
+    const nodes = getNodes(result.data);
+    allNodes = [...allNodes, ...nodes];
+    body.variables.offset += graphqlPageSize;
+    if (nodes.length < graphqlPageSize) {
+      return allNodes;
+    } else {
+      // sleep for a while
+      await new Promise(resolve => setTimeout(resolve, 100));
```

Comment:
Does it fail without this?

Reply:
Oh I didn't change this part, not sure yet